src/main/java/javaslang/collection/Array.java

```diff
@@ -203,6 +203,41 @@ private Array(Object[] back) {
         return Array.ofAll(Iterator.ofAll(array));
     }
 
+    /**
+     * Returns an Array containing {@code n} values of a given Function {@code f}
+     * over a range of integer values from 0 to {@code n - 1}.
+     *
+     * @param <T> Component type of the Array
+     * @param n The number of elements in the Array
+     * @param f The Function computing element values
+     * @return An Array consisting of elements {@code f(0),f(1), ..., f(n - 1)}
+     * @throws NullPointerException if {@code f} is null
+     */
+    public static <T> Array<T> tabulate(int n, Function<Integer, ? extends T> f) {
+        Objects.requireNonNull(f, "f is null");
+        int nOrZero = java.lang.Math.max(n, 0);
+        @SuppressWarnings("unchecked")
```

Comment:
<code>Array::wrap()</code> accept <code>Object[]</code> as parameter, so you can do this:
```java
Object[] elements = new Object[nOrZero];
```